### PR TITLE
L2regularizer

### PR DESCRIPTION
**Pull request by wadaniel**
added l2 regularization for NN weights update,
added DQN,
work on CMAIRL ,
minor cleanups,
fixed loading of RL runs,
includes some changes from gfpt-3 branch

### CODE REVIEW

File: tools/rlview/__main__.py [hunk 1]
```diff
@@ -79,11 +76,10 @@ def plotRewardHistory(ax, dirs, results, minReward, maxReward, averageDepth, max
   confIntervalHistory = np.array(confIntervalHistory)
 
   # Plotting common plot
-    
+  clr='red'
+  if ('GFPT' in dirs[resId]): clr='blue'    
   epList = range(0, len(rewardHistory)) 
-  ax.plot(epList, rewardHistory, 'x', markersize=1, label='Episode Reward (' + dirs[resId] + ')')
-  ax.plot(epList, meanHistory, '-', label=str(averageDepth) + '-Episode Average (' + dirs[resId] + ')')
-  ax.fill_between(epList, (meanHistory-confIntervalHistory), (meanHistory+confIntervalHistory), color='b', alpha=.1, label='98% Confidence Interval (' + dirs[resId] + ')')
```
Comment (webepasc) [inline tools/rlview/__main__.py:86]: why no-more errorbars and rewards?

**Review by webepasc — APPROVED** [review]

File: tools/rlview/__main__.py (hunk 1 above)
Comment (wadaniel) [reply tools/rlview/__main__.py:86]: changes in "sergios modules" are there because I updated my branch with his gfpt-3 branch

Comment (webepasc) [reply tools/rlview/__main__.py:86]: kk

File: source/modules/neuralNetwork/neuralNetwork._cpp [hunk 2]
```diff
@@ -100,13 +100,6 @@ void NeuralNetwork::initialize()
   _outputValues.resize(N);
   for (size_t b = 0; b < N; b++) _outputValues[b].resize(OC);
 
-  // Allocating storage for formatted input gradients (only for training)
```
Comment (SergioMartin86) [inline source/modules/neuralNetwork/neuralNetwork._cpp:103]: Please put it back, needed for ddpg

File: source/modules/neuralNetwork/neuralNetwork._hpp [hunk 3]
```diff
@@ -57,11 +57,6 @@ class NeuralNetwork : public Module
    */
   std::vector<float> _rawInputGradients;
 
-  /**
-   * @brief Formatted data for the NN input gradients. Format: NxIC (N: Mini-batch size, IC: Input channels).
```
Comment (SergioMartin86) [inline source/modules/neuralNetwork/neuralNetwork._hpp:61]: put back for ddpg

File: source/modules/solver/agent/agent.config [hunk 4]
```diff
@@ -60,6 +60,11 @@
     "Type": "float",
     "Description": "The learning rate to use for the NN hyperparameter optimizer."
   },
+  {
+   "Name": [ "L2 Regularization" ],
```
Comment (SergioMartin86) [inline source/modules/solver/agent/agent.config:64]: make more descriptive, either with 2 settings (on/off + magnitude) or say "L2 Regularization Importance"

**Review by SergioMartin86 — CHANGES_REQUESTED** [review]

File: source/modules/solver/agent/agent.config (hunk 4 above)
Comment (wadaniel) [reply source/modules/solver/agent/agent.config]: done

File: source/modules/neuralNetwork/neuralNetwork._hpp (hunk 3 above)
Comment (wadaniel) [reply source/modules/neuralNetwork/neuralNetwork._hpp:61]: done

**Review by SergioMartin86 — APPROVED** [review]

**Review by webepasc — APPROVED** [review]

**Pull request merged**